### PR TITLE
docs: Improve Getting Started

### PR DESCRIPTION
**Pull request by ThangHuuVu**
## ☕️ Reasoning

Improve Getting Started Page

## 🧢 Checklist

- [x] Introduction - Mention framework & runtime agnostic
- [ ] OAuth tutorial: Trying to reuse most material from existing guide
  - [ ] SolidStart
  - [ ] SvelteKit
  - [x] Next

## 📌 Resources

- [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md)
- [Contributing guidelines](https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md)
- [Code of conduct](https://github.com/nextauthjs/.github/blob/main/CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

File: docs/docs/getting-started/02-oauth-tutorial.mdx
Comment (balazsorban44) [inline docs/docs/getting-started/02-oauth-tutorial.mdx:64]:
```suggestion
can be handled by NextAuth.js. In this way, NextAuth.js stays in charge of the whole application's authentication request/response flow.
```

File: docs/docs/getting-started/02-oauth-tutorial.mdx
Comment (balazsorban44) [inline docs/docs/getting-started/02-oauth-tutorial.mdx:66]:
```suggestion
NextAuth.js is fully customizable - [our guides section](/guides/overview) teaches you how to set it up to handle auth in different ways. All the possible configuration options are [listed here](/reference/configuration/auth-config).
```

File: docs/docs/getting-started/02-oauth-tutorial.mdx
Comment (balazsorban44) [inline docs/docs/getting-started/02-oauth-tutorial.mdx:87]: `useSession` wasn't mentioned at this point, we should explain in a sentence what that is.

File: docs/docs/getting-started/02-oauth-tutorial.mdx
Comment (balazsorban44) [inline docs/docs/getting-started/02-oauth-tutorial.mdx:104]: Move this up to satisfy https://github.com/nextauthjs/next-auth/pull/6551/files#r1089949056

File: docs/docs/getting-started/02-oauth-tutorial.mdx
Comment (balazsorban44) [inline docs/docs/getting-started/02-oauth-tutorial.mdx:112]:
```suggestion
NextAuth.js exposes a [`useSession()`](/reference/react/#usesession) React Hook so that you can check if someone is logged in:
```

File: docs/docs/getting-started/02-oauth-tutorial.mdx
Comment (balazsorban44) [inline docs/docs/getting-started/02-oauth-tutorial.mdx:144]: Could drop a link to https://reactjs.org/docs/context.html

File: docs/docs/getting-started/02-oauth-tutorial.mdx
Comment (balazsorban44) [inline docs/docs/getting-started/02-oauth-tutorial.mdx:148]: Let's rephrase to remove "easy". Nothing is if you are just starting out! :green_heart:

File: docs/docs/getting-started/02-oauth-tutorial.mdx
Comment (balazsorban44) [inline docs/docs/getting-started/02-oauth-tutorial.mdx:152]:
```suggestion
import authOptions from "../auth/[...nextauth]"
```

Let's not assume anything about the user's setup. They can change it if they want to.

File: docs/docs/getting-started/02-oauth-tutorial.mdx
Comment (balazsorban44) [inline docs/docs/getting-started/02-oauth-tutorial.mdx:183]:
```suggestion
Ok, we have our app set up with NextAuth.js, however, if you run the app right now, it won't work as we haven't configured our OAuth provider (**GitHub**) yet.
```

File: docs/docs/getting-started/02-oauth-tutorial.mdx
Comment (balazsorban44) [inline docs/docs/getting-started/02-oauth-tutorial.mdx:16]: We should have our own OAuth page under Concepts, taking the graph from https://next-auth.js.org/configuration/providers/oauth

File: docs/docs/getting-started/02-oauth-tutorial.mdx
Comment (balazsorban44) [inline docs/docs/getting-started/02-oauth-tutorial.mdx:189]:
```suggestion
We need to register our new app in GitHub, so that when NextAuth.js forwards the authorization requests to it, GitHub can recognize your application and prompt the user to sign in.
```

File: docs/docs/getting-started/02-oauth-tutorial.mdx
Comment (balazsorban44) [inline docs/docs/getting-started/02-oauth-tutorial.mdx:193]:
```suggestion
Log in to **GitHub**, go to [`Settings / Developers / OAuth Apps`](https://github.com/settings/developers) and click "New OAuth App"
```

File: docs/docs/getting-started/02-oauth-tutorial.mdx
Comment (balazsorban44) [inline docs/docs/getting-started/02-oauth-tutorial.mdx:215]:
```suggestion
NextAuth.js will already create this API endpoint for you when we start the application later. Note that because we're using Next.js, locally it starts our server on port `3000` by default. Hence, the origin is `http://localhost:3000`.
```

File: docs/docs/getting-started/02-oauth-tutorial.mdx
Comment (balazsorban44) [inline docs/docs/getting-started/02-oauth-tutorial.mdx:213]: Let's add a note here where the last portion of the URL (`github`) comes from, and what is the relation with `signIn("github")`.

Not sure what's the best way to document this, but it's the provider configuration's `id` property.

File: docs/docs/getting-started/02-oauth-tutorial.mdx
Comment (balazsorban44) [inline docs/docs/getting-started/02-oauth-tutorial.mdx:239]:
```suggestion
Keep both your Client ID and Client Secret secure and never expose them to the public or share them with people outside your organization. With them, a malicious actor could hijack your application and cause you and your user serious problems!
```

File: docs/docs/getting-started/02-oauth-tutorial.mdx
Comment (balazsorban44) [inline docs/docs/getting-started/02-oauth-tutorial.mdx:245]:
```suggestion
As noted previously, NextAuth.js has built-in support for multiple OAuth providers, <a href="">here is the full list</a>. You can also easily build your own in case the provider you need is not on the list.
```

Also, missing link?

File: docs/docs/getting-started/02-oauth-tutorial.mdx
Comment (balazsorban44) [inline docs/docs/getting-started/02-oauth-tutorial.mdx:308]:
```suggestion
Once inserted and correct, GitHub will redirect the user to our app and NextAuth.js will take care of any further calls with GitHub to get access to the user profile and start a user session safely in the background:
```

File: docs/docs/getting-started/02-oauth-tutorial.mdx
Comment (balazsorban44) [inline docs/docs/getting-started/02-oauth-tutorial.mdx:316]: I think this is unnecessary to mention here.

File: docs/docs/getting-started/02-oauth-tutorial.mdx
Comment (ThangHuuVu) [reply docs/docs/getting-started/02-oauth-tutorial.mdx:148]: Totally!

File: docs/docs/getting-started/02-oauth-tutorial.mdx
Comment (ThangHuuVu) [reply docs/docs/getting-started/02-oauth-tutorial.mdx:16]: We have that page https://authjs.dev/concepts/oauth Are you suggesting to link it here?

File: docs/docs/getting-started/02-oauth-tutorial.mdx
Comment (lluia) [inline docs/docs/getting-started/02-oauth-tutorial.mdx:73]:
```suggestion
can be handled by Auth.js. In this way, Auth.js stays in charge of the whole application's authentication request/response flow.
```

File: docs/docs/getting-started/02-oauth-tutorial.mdx
Comment (lluia) [inline docs/docs/getting-started/02-oauth-tutorial.mdx:75]:
```suggestion
Auth.js is fully customizable - [our guides section](/guides/overview) teaches you how to set it up to handle auth in different ways. All the possible configuration options are [listed here](/reference/configuration/auth-config).
```

File: docs/docs/getting-started/02-oauth-tutorial.mdx
Comment (lluia) [inline docs/docs/getting-started/02-oauth-tutorial.mdx:155]:
```suggestion
To protect your API Routes (blocking unauthorized access to resources), you can use [`getServerSession()`](/reference/nextjs#getserversession) to know whether a session exists or not:
```

File: docs/docs/getting-started/02-oauth-tutorial.mdx
Comment (lluia) [inline docs/docs/getting-started/02-oauth-tutorial.mdx:189]: I guess this should be addressed before we merge right?

File: docs/docs/getting-started/02-oauth-tutorial.mdx
Comment (lluia) [inline docs/docs/getting-started/02-oauth-tutorial.mdx:193]:
```suggestion
Ok, we have our app set up with Auth.js; however, if you run the app right now, it won't work as we haven't configured our OAuth provider (**GitHub**) yet.
```

File: docs/docs/getting-started/02-oauth-tutorial.mdx
Comment (lluia) [inline docs/docs/getting-started/02-oauth-tutorial.mdx:199]:
```suggestion
We need to register our new app in GitHub, so that when Auth.js forwards the authorization requests to it, GitHub can recognize your application and prompt the user to sign in.
```

File: docs/docs/getting-started/02-oauth-tutorial.mdx
Comment (lluia) [inline docs/docs/getting-started/02-oauth-tutorial.mdx:225]:
```suggestion
Auth.js will already create this API endpoint for you when we start the application later. Note that because we're using Next.js, locally it starts our server on port `3000` by default. Hence, the origin is `http://localhost:3000`.
```

File: docs/docs/getting-started/02-oauth-tutorial.mdx
Comment (lluia) [inline docs/docs/getting-started/02-oauth-tutorial.mdx:262]:
```suggestion
As noted previously, Auth.js has built-in support for multiple OAuth providers, <a href="">here is the full list</a>. You can also easily build your own if the provider you need is not on the list.
```

File: docs/docs/getting-started/02-oauth-tutorial.mdx
Comment (lluia) [inline docs/docs/getting-started/02-oauth-tutorial.mdx:328]:
```suggestion
Once inserted and correct, GitHub will redirect the user to our app and Auth.js will take care of any further calls with GitHub to get access to the user profile and start a user session safely in the background:
```

**Review by 0ubbe — COMMENTED** [review]
Awesome work 🔥

Small suggestions to update `NextAuth.js` → `Auth.js` to stay consistent.

**Comment by lluia** [comment]
Oh I see now the change `Auth.js → NextAuth.js` was intentional, removing my comments then 😅

File: docs/docs/getting-started/02-oauth-tutorial.mdx
Comment (balazsorban44) [inline docs/docs/getting-started/02-oauth-tutorial.mdx]:
```suggestion
The easiest way is to set up Auth.js with an [OAuth](/concepts/oauth) provider. In this tutorial, we'll be setting Auth.js in an application with a framework of your choice to be able to log in with **GitHub**.
```

**Review by ndom91 — APPROVED** [review]
LGTM :+1:

**Review by balazsorban44 — APPROVED** [review]

**Pull request merged**
